examples/chef/common/chef-resource-monitoring-delegates.cpp

```diff
@@ -0,0 +1,172 @@
+    return Status::Success;
+}
+
+void HepaFilterMonitoring::Shutdown()
+{
+    if (gHepaFilterInstance != nullptr)
+    {
+        delete gHepaFilterInstance;
+        gHepaFilterInstance = nullptr;
+    }
+    if (gHepaFilterDelegate != nullptr)
+    {
+        delete gHepaFilterDelegate;
+        gHepaFilterDelegate = nullptr;
+    }
+}
+
+void emberAfActivatedCarbonFilterMonitoringClusterInitCallback(chip::EndpointId endpoint)
+{
+    VerifyOrDie(gActivatedCarbonFilterInstance == nullptr && gActivatedCarbonFilterDelegate == nullptr);
+    gActivatedCarbonFilterDelegate = new ActivatedCarbonFilterMonitoringDelegate;
+    gActivatedCarbonFilterInstance = new ResourceMonitoring::Instance(
+        gActivatedCarbonFilterDelegate, endpoint, ActivatedCarbonFilterMonitoring::Id,
+        static_cast<uint32_t>(gActivatedCarbonFeatureMap.to_ulong()), ResourceMonitoring::DegradationDirectionEnum::kDown, true);
+    gActivatedCarbonFilterInstance->Init();
+}
+
+void emberAfHepaFilterMonitoringClusterInitCallback(chip::EndpointId endpoint)
+{
+    VerifyOrDie(gHepaFilterInstance == nullptr && gHepaFilterDelegate == nullptr);
+
+    gHepaFilterDelegate = new HepaFilterMonitoringDelegate;
+    gHepaFilterInstance = new ResourceMonitoring::Instance(gHepaFilterDelegate, endpoint, HepaFilterMonitoring::Id,
+                                                           static_cast<uint32_t>(gHepaFilterFeatureMap.to_ulong()),
+                                                           ResourceMonitoring::DegradationDirectionEnum::kDown, true);
+    gHepaFilterInstance->Init();
+}
+
+CHIP_ERROR ImmutableReplacementProductListManager::Next(ReplacementProductStruct & item)
+{
+    if (mIndex >= kReplacementProductListMaxSize)
+    {
+        return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+    }
+
+    switch (mIndex)
+    {
+    case 0: {
+        item.SetProductIdentifierType(ResourceMonitoring::ProductIdentifierTypeEnum::kUpc);
+        item.SetProductIdentifierValue(CharSpan::fromCharString("111112222233"));
+        break;
+    case 1:
+        item.SetProductIdentifierType(ResourceMonitoring::ProductIdentifierTypeEnum::kGtin8);
+        item.SetProductIdentifierValue(CharSpan::fromCharString("gtin8xxx"));
+        break;
+    case 2:
+        item.SetProductIdentifierType(ResourceMonitoring::ProductIdentifierTypeEnum::kEan);
+        item.SetProductIdentifierValue(CharSpan::fromCharString("4444455555666"));
+        break;
+    case 3:
+        item.SetProductIdentifierType(ResourceMonitoring::ProductIdentifierTypeEnum::kGtin14);
+        item.SetProductIdentifierValue(CharSpan::fromCharString("gtin14xxxxxxxx"));
+        break;
+    case 4:
+        item.SetProductIdentifierType(ResourceMonitoring::ProductIdentifierTypeEnum::kOem);
+        item.SetProductIdentifierValue(CharSpan::fromCharString("oem20xxxxxxxxxxxxxxx"));
+        break;
+    default:
+        return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+        break;
+    }
+    }
```

Comment:
what is up with these extra braces? Does this compile? this seems to match the `case 0` brace, however why does the encompass several case statements?

Reply:
After you make all the code fixes, please confirm that you compiled and tested this example.

Reply:
We should probably figure out a way to compile all the classes at least even if we do not compile all examples (because it is slow) ... it seems painful to maintain things if we have no CI that validates that at least things compile.

Reply:
Unnecessary braces removed. PTAL.

Reply:
I think the current `all-cluster-app` will compile all clusters (or classes)?  BTW, I think maybe one day Chef could also make a all-cluster-app.